### PR TITLE
[TwigBridge] Fix workflow test with deps=low

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | none
| License       | MIT
| Doc PR        | none

In the original PR the tests were not failling but we should not use the new feature with deps=low until it fetch the branch with the new feature. So this fix the test for the WorkflowExtension in TwigBridge.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Should we plan to remove all the new `@legacy` method? Won't that reduce the test coverage?
Shouldn't we instead upgrade the test suite to use the not-deprecated API (and keep a few legacy call to just to ensure the legacy way keeps working)?
Feels like so to me.

**Comment by Simperfit** [comment]
we do not plan to remove but to update when we will update the Bridge to
use the new features. but yeah I agree if we can update now let’s do it :)
Le mar. 12 déc. 2017 à 11:08, Nicolas Grekas <notifications@github.com> a
écrit :

**Comment by chalasr** [comment]
Keeping these legacy tests do not make sense to me either. `setUp()` should just use the right api based on method existence (the actual code does not even use this part of the api, only the test does).

**Comment by Simperfit** [comment]
Oh, maybe we should add a condition like you sais and instead of using the
annotation use the method from phpunit ?
Le mar. 12 déc. 2017 à 11:16, Robin Chalas <notifications@github.com> a
écrit :

**Comment by chalasr** [comment]
IMO there is no deprecation to assert here. 
We want to keep 3.x compatibility, the test should just use the right api i.e. `$registry->{method_exists($registry, $_ = 'addWorkflow') ? $_ : 'add'}($workflow);`, no possible deprecation as the new api is used if it exists.

**Comment by Simperfit** [comment]
We can remove the deprecation tag ;)

File: src/Symfony/Bridge/Twig/Tests/Extension/WorkflowExtensionTest.php [hunk 1]
```diff
@@ -57,25 +38,10 @@ protected function setUpLegacyAdd()
         $workflow = new Workflow($definition);
 
         $registry = new Registry();
-        $registry->add($workflow, new InstanceOfSupportStrategy(\stdClass::class));
-
+        $registry->{method_exists($registry, $_ = 'addWorkflow') ? $_ : 'add'}($workflow, new InstanceOfSupportStrategy(\stdClass::class));
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Tests/Extension/WorkflowExtensionTest.php]: $addWorkflow = method_exists($registry, 'addWorkflow') ? 'addWorkflow' : 'add';
$registry->$addWorkflow($workflow, new InstanceOfSupportStrategy(\stdClass::class));

File: src/Symfony/Bridge/Twig/Tests/Extension/WorkflowExtensionTest.php [hunk 2]
```diff
@@ -57,25 +37,14 @@ protected function setUpLegacyAdd()
         $workflow = new Workflow($definition);
 
         $registry = new Registry();
-        $registry->add($workflow, new InstanceOfSupportStrategy(\stdClass::class));
-
+        $addWorkflow = method_exists($registry, 'addWorkflow') ? 'addWorkflow' : 'add';
+        $supportStrategy = class_exists('Symfony\Component\Workflow\SupportStrategy\InstanceOfSupportStrategy')
+            ? new \Symfony\Component\Workflow\SupportStrategy\InstanceOfSupportStrategy(\stdClass::class)
+            : new \Symfony\Component\Workflow\SupportStrategy\ClassInstanceSupportStrategy(\stdClass::class);
```
Comment (chalasr) [inline src/Symfony/Bridge/Twig/Tests/Extension/WorkflowExtensionTest.php]: please keep using use statements :)

**Comment by Simperfit** [comment]
🍾 it's green !

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Pull request merged**